plugins/overlay.cpp

```diff
@@ -0,0 +1,80 @@
+#include "df/viewscreen_titlest.h"
+
+#include "modules/Gui.h"
+
+#include "PluginManager.h"
+#include "VTableInterpose.h"
+#include "uicommon.h"
+
+using namespace DFHack;
+
+DFHACK_PLUGIN("overlay");
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+REQUIRE_GLOBAL(enabler);
+REQUIRE_GLOBAL(gps);
+
+static const std::string button_text = "[ DFHack Launcher ]";
+static bool clicked = false;
+
+static bool handle_click() {
+    int32_t x, y;
+    if (!enabler->tracking_on || !enabler->mouse_lbut || clicked ||
+            !Gui::getMousePos(x, y))
+        return false;
+    if (y == gps->dimy - 1 && x >= 1 && (size_t)x <= button_text.size()) {
+        clicked = true;
+        Core::getInstance().setHotkeyCmd("gui/launcher");
+        return true;
+    }
+    return false;
+}
+
+static void draw_widgets() {
+    int x = 1;
+    int y = gps->dimy - 1;
+    OutputString(COLOR_GREEN, x, y, button_text);
+}
+
+struct title_hook : df::viewscreen_titlest {
+    typedef df::viewscreen_titlest interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(void, feed, (set<df::interface_key> *input)) {
+        if (!handle_click())
+            INTERPOSE_NEXT(feed)(input);
+    }
+    DEFINE_VMETHOD_INTERPOSE(void, render, ()) {
+        INTERPOSE_NEXT(render)();
+        draw_widgets();
```

Comment:
I would swap the order of these, in case screens are drawing something important in this region

Reply:
I tested that, and nothing is shown at all if the order is reversed. At least on the title screen.

Reply:
Bah, it must be drawing blank characters on the whole screen. Any screen using drawBorder() would run into the same thing, probably.

Not sure what a good fix is, then. Maybe scanning for text characters before writing? I don't want to overwrite things like the weather indicator.

Reply:
I planned to move the weather indicator somewhere non-conflicting, like a few tiles to the right. Are there others that I need to watch out for?

Reply:
done, but you have to `rm dfhack-config/dwarfmonitor.json` to get the update to the weather indicator position.

Reply:
I guess moving that is fine. I noticed the controls on `gui/load-screen` when I was loading a game. That one won't be covered if we are skipping DFHack screens, though.

I don't know of any DF screens that would conflict off the top of my head - most draw a border, which we can draw on top of easily enough. Exceptions that I know of: title, options, save, load, clean